app/api/v0/proofs/download/all/[block]/route.ts

```diff
@@ -0,0 +1,66 @@
+import AdmZip from "adm-zip"
+
+import { db } from "@/db"
+
+export async function GET(
+  _request: Request,
+  { params }: { params: { block: string } }
+) {
+  const { block } = params
+
+  const proofRows = await db.query.proofs.findMany({
+    columns: {
+      block_number: true,
+      cluster_id: true,
+      team_id: true,
+      proof_id: true,
+    },
+    with: {
+      proof_binary: true,
+    },
+    where: (proofs, { eq }) => eq(proofs.block_number, Number(block)),
+  })
+
+  if (!proofRows || !proofRows.length) {
+    return new Response("No proofs found", { status: 404 })
+  }
+
+  const binaryBuffers: { binaryBuffer: Buffer; filename: string }[] = []
+
+  for (const proofRow of proofRows) {
+    const team = await db.query.teams.findFirst({
+      columns: {
+        name: true,
+      },
+      where: (teams, { eq }) => eq(teams.id, proofRow.team_id),
+    })
+
+    const teamName = team?.name ? team.name : proofRow.cluster_id.split("-")[0]
+    const filename = `block_${block}_proof_${proofRow.proof_id}_${teamName}.txt`
```

Comment:
Added `proof_id` to this since at least during testing I ended up with two different proofs from the same team that had ended up with the exact same name, so only the last one persisted.

In production, I'm not sure this is going to be an issue and I don't love using `proof_id` since it's useless to the user.

It would be nice to show the `proof_type` here instead, which is nested in the cluster information. Drizzle is new to me so not sure immediately how to access that here, but can figure it out if we like the idea

Reply:
Okay that was really easy, so I pushed a commit (9691de1) using `proof_type` and `cycle_type` in the proof filename instead of `proof_${proof_id}`